bioamptool.py

```diff
@@ -221,104 +246,152 @@ def log_ten_minute_data(verbose=False):
-                if elapsed_since_last_10_minutes >= 600:  # Check if 10 minutes have passed
-                    log_ten_minute_data(verbose)  # Log data for the last 10 minutes
+                if elapsed_time >= 1:  
+                    log_one_second_data(verbose)
+                    start_time = current_time
+
+                if elapsed_since_last_10_minutes >= 600:
+                    log_ten_minute_data(verbose) 
+
                 if gui_flag:
-                    QtWidgets.QApplication.processEvents()  # Process GUI events if GUI is enabled
-                    
-                if msvcrt.kbhit() and msvcrt.getch() == b'q':  # Exit the loop if 'q' is pressed
-                    ser.write(b'STOP\n')
-                    print("Process interrupted by user")
-                    break
+                    QtWidgets.QApplication.processEvents()
 
-        except KeyboardInterrupt:
-            ser.write(b'STOP\n')
-            print("Process interrupted by user")
+                if run_time and current_time >= end_time:
+                    print("Runtime Over, sending STOP command...")
+                    send_command(ser, 'STOP')
+                    break
 
-        finally:
-            if csv_file:
-                csv_file.close()
-                print(f"CSV recording saved as {csv_filename}")
-            print(f"Exiting.\nTotal missing samples: {missing_samples}")  # Print final missing samples count
+    except KeyboardInterrupt:
+        print("Process interrupted by user")
+    
+    finally:
+        if ser.is_open:
+            send_command(ser, 'STOP')
+            ser.reset_input_buffer()
+            ser.reset_output_buffer()
+            ser.close()
+
+        if lsl_outlet:     # Assuming LSL outlet cleanup if required
+            lsl_outlet = None
         
+        if csv_file:    # Ensure CSV file is closed
+            csv_file.close()
+            print(f"CSV recording saved as {csv_filename}")
+        
+        if gui_flag:    # Ensure GUI is properly closed
+            if timer:
+                timer.stop()
+            if app:
+                app.quit() 
+
+    print(f"Exiting.\nTotal missing samples: {missing_samples}")
+    sys.exit(0)
+
+def signal_handler(sig, frame):
+    print("Process interrupted by user")
+    global ser, csv_file, lsl_outlet, app  # Declare global variables
+    
+    ser = None
+    csv_file = None
+    lsl_outlet = None
+    app = None
+    
+    # Perform any necessary cleanup here
+    if ser and ser.is_open:
+        send_command(ser, 'STOP')
+        ser.reset_input_buffer()
+        ser.reset_output_buffer()
+        ser.close()
+    if lsl_outlet:
+        lsl_outlet = None  # Cleanup if needed
+    if csv_file:
+        csv_file.close()  # Close the CSV file
+        print(f"CSV recording saved as {csv_filename}")
+    if app:
+        app.quit()  # Close the GUI application
+    sys.exit(0)
```

Comment:
Why are you re-doing things using the same code? what's the meaning of this cleanup?

Reply:
Now, it is resolved.